CondTools/RPC/src/RPCDCCLinkMapHandler.cc

```diff
@@ -0,0 +1,116 @@
+#include "CondTools/RPC/interface/RPCDCCLinkMapHandler.h"
+
+#include <fstream>
+#include <sstream>
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/Exception.h"
+
+#include "CondCore/CondDB/interface/ConnectionPool.h"
+#include "CondCore/CondDB/interface/Types.h"
+
+#include "RelationalAccess/ICursor.h"
+#include "RelationalAccess/IQuery.h"
+#include "RelationalAccess/IQueryDefinition.h"
+#include "RelationalAccess/ISchema.h"
+#include "RelationalAccess/ISessionProxy.h"
+
+#include "CoralBase/Attribute.h"
+#include "CoralBase/AttributeList.h"
+
+#include "CondTools/RPC/interface/RPCLBLinkNameParser.h"
+
+RPCDCCLinkMapHandler::RPCDCCLinkMapHandler(edm::ParameterSet const & _config)
+    : id_(_config.getParameter<std::string>("identifier"))
+    , data_tag_(_config.getParameter<std::string>("dataTag"))
+    , since_run_(_config.getParameter<unsigned long long>("sinceRun"))
+    , txt_file_(_config.getUntrackedParameter<std::string>("txtFile", ""))
+{
+    cond::persistency::ConnectionPool _connection;
+    _connection.setParameters(_config.getParameter<edm::ParameterSet>("DBParameters"));
+    _connection.configure();
+    edm::LogInfo("RPCDCCLinkMapHandler") << "Opening Input Session";
+    input_session_ = _connection.createSession(_config.getParameter<std::string>("connect"));
```

Comment:
What is the reason to create session and transaction in the constructor ( and close them in the destructor )?
It looks to me that the queries on the db are:
- involving OMDS tables
- read only
- all performed in the getNewObjects method
It would be better to get directly the coral session from the pool ( createCoralSession ), open the normal transaction ( TransactionScope is required only for updating transaction ) in the getNewObjects. In this way, the session and transaction lifetime are better defined and probably minimised.